docs/blueprint/EP039.rst

```diff
@@ -0,0 +1,280 @@
+      - The defaut is `event_logs`.
+      - Kafka ensures these events are durably stored and available for consumption by downstream services.
+  - Production
+    - Events are sent in batches asynchronously, ensuring maximum efficiency in network requests and processing.
+  - Key Components
+    - KafkaSendOperations: Handles the creation of Kafka topics (if necessary) and manages event publishing.
+
+
+IV. Configurations in settings.py
+==============================
+
+IV. Configurations in settings.py
+
+1. This section describes the key configuration parameters used by the kafka_napp application. These constants define how the application connects to Kafka, manages topics, handles message delivery, and filters events.
+
+- Key Configuration Constants
+  - BOOTSTRAP_SERVERS
+    - Description: Specifies the Kafka server's address (hostname and port).
+    - Usage: This setting is critical for establishing the connection with the Kafka broker.
+  - ACKS
+    - Description: Determines the level of message acknowledgements required by Kafka.
+    - Available Options:
+      - 0: No acknowledgements required.
+      - 1: Only the leader broker must acknowledge.
+      - "all": All in-sync replicas must acknowledge the message.
+      - Usage: Controls the reliability and durability of message delivery.
+      - Note: This may increase latency between the application during sends. To add this functionality, ensure tolerance for when the application is sending to kafka but the also wants to enqueue new requests.
+  - DEFAULT_NUM_PARTITIONS
+    - Description: Sets the default number of partitions per Kafka topic.
+    - Usage: Influences the parallelism and throughput of message processing.
+  - IGNORED_EVENTS
+    - Description: Defines a set of event names that should be filtered out and not processed.
+    - Usage: Helps in ignoring events that are not relevant to the application's processing logic.
+  - REPLICATION_FACTOR
+    - Description: Specifies the number of times each partition is replicated across Kafka brokers.
+    - Usage: Enhances data redundancy and fault tolerance.
+    - Note: The value should not exceed the number of available Kafka brokers.
+  - TOPIC_NAME
+    - Description: The Kafka topic from which messages are sent and/or consumed.
+    - Usage: Acts as the central channel for event logging and communication between components.
+  - COMPRESSION_TYPE
+    - Description: The type of compression applied to messages sent to Kafka.
+    - Example: "gzip"
+    - Usage: Reduces network load and improves performance by compressing message data.
+
+
+V. Events
+==========
+
+  1. kafka_napp listens to all possible events. In addition, it does not currently ignore any events. If it starts ignoring events, they will be labeled here.
+
+
+VI. REST API
+=============
+
+  - ``GET /api/kytos/kafka/v1/``: Currently does nothing. Could be used to gather metrics on the NApp, like I/O, errors, etc.
```

Comment:
If the endpoints currently don't do anything, then they should be removed. We should only implement endpoints which a clear purpose and functionality in mind. I recommend you leave in the future work section to leave the endpoints there, and then in the future consider what will be implemented with the use cases in mind.